src/vendor/azure-uamqp-c/src/link.c

```diff
@@ -1186,6 +1186,65 @@ int link_set_max_link_credit(LINK_HANDLE link, uint32_t max_link_credit)
     return result;
 }
 
+int link_reset_link_credit(LINK_HANDLE link, uint32_t link_credit, bool drain)
+{
+    int result;
+    FLOW_HANDLE flow;
+
+    if (link == NULL)
+    {
+        result = __FAILURE__;
+    }
```

Comment:
We may need to add a check here to confirm this is a receiver link - as this operation is not legal for a sender link.